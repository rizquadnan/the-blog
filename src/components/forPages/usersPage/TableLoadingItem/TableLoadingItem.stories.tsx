import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { TableLoadingItem } from './TableLoadingItem'
import { Table, Tbody, Th, Thead } from '@chakra-ui/react'

type TTableLoadingItem = React.ComponentProps<typeof TableLoadingItem>
export default {
  title: 'Feedback/TableLoadingItem',
  component: TableLoadingItem,
} as Meta<TTableLoadingItem>

const Template: Story<TTableLoadingItem> = () => (
  <Table variant="striped">
    <Thead>
      <Th>1</Th>
      <Th>2</Th>
      <Th>3</Th>
      <Th>4</Th>
      <Th>5</Th>
      <Th>6</Th>
    </Thead>
    <Tbody>
      <TableLoadingItem />
    </Tbody>
  </Table>
)

export const Default = Template.bind({})
