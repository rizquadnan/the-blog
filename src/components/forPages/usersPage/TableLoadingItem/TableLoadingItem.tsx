import { SkeletonText, Td, Tr, useColorModeValue } from '@chakra-ui/react'

export function TableLoadingItem() {
  const skeletonBackground = useColorModeValue('gray.500', 'black')

  return (
    <Tr>
      {Array(6)
        .fill('')
        .map((_, index) => (
          <Td key={index}>
            <SkeletonText
              noOfLines={1}
              w="100%"
              startColor={skeletonBackground}
              endColor={skeletonBackground}
            />
          </Td>
        ))}
    </Tr>
  )
}
