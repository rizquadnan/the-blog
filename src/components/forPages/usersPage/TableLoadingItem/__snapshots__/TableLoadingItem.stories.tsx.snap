// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback/TableLoadingItem Default smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "1",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "2",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "3",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "4",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "5",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "6",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "feedback-tableloadingitem--default",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
